larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/ShowerESTAREnergy_tool.cc

```diff
@@ -0,0 +1,293 @@
+//############################################################################
+//### Name:        ShowerESTAREnergy                                       ###
+//### Author:      Tom Ham                                                 ###
+//### Date:        06/09/2021                                              ###
+//### Description: Tool for finding the energy of a shower by using        ###
+//###              the ESTAR database.                                     ###
+//###              Derived from the method used by ArgoNeuT.               ###
+//############################################################################
+
+//Framework Includes
+#include "art/Utilities/ToolMacros.h"
+
+//LArSoft Includes
+#include "larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/IShowerTool.h"
+#include "lardataobj/RecoBase/Cluster.h"
+#include "larcoreobj/SimpleTypesAndConstants/PhysicalConstants.h"
+#include "larreco/Calorimetry/CalorimetryAlg.h"
+
+// ROOT includes
+#include "TFile.h"
+#include "TGraph2D.h"
+
+namespace ShowerRecoTools {
+
+  class ShowerESTAREnergy:IShowerTool {
+
+  public:
+
+  ShowerESTAREnergy(const fhicl::ParameterSet& pset);
+    
+  ~ShowerESTAREnergy(); 
+        
+  //Physics Function. Calculate the shower Energy.
+  int CalculateElement(const art::Ptr<recob::PFParticle>& pfparticle,
+  art::Event& Event,
+  reco::shower::ShowerElementHolder& ShowerElementHolder
+  ) override;
+
+  private:
+    
+  double CalculateEnergy(const detinfo::DetectorClocksData& clockData,
+           const detinfo::DetectorPropertiesData& detProp, 
+           const std::vector<art::Ptr<recob::Hit> >& hits,
+           const geo::PlaneID::PlaneID_t plane);
+
+  art::InputTag fPFParticleLabel;
+  int fVerbose;
+
+  //Services
+  art::ServiceHandle<geo::Geometry> fGeom;
+  calo::CalorimetryAlg              fCalorimetryAlg;    
+
+  //fcl params 
+  std::string fShowerEnergyOutputLabel;
+  std::string fShowerBestPlaneOutputLabel; 
+  bool fSCECorrectEField;
+
+  double nominal_Efield = 0;         // Nominal E-field in the detector
+  double localEfield = 0;            // Local E-field at the geometric centre of the shower 
+  double localEfield_cweighted = 0;  // Local E-field at the charge weighted centre of the shower
+
+  std::vector<std::tuple<int, double>> sp_hits_tuple; // tuple holding the key and local E-field for all the sp hits 
+
+  TGraph2D *t_estar;
```

Comment:
@PandoraPFA @AndyChappell we may have to think about how to handle root dependencies here...

Reply:
As discussed elsewhere, there are already ROOT dependencies in LArPandoraShower, so this should be fine